client/boot/common.js

```diff
@@ -437,10 +433,55 @@ const boot = ( currentUser, registerRoutes ) => {
 	} );
 };
 
-export const bootApp = ( appName, registerRoutes ) => {
+function waitForCookieAuth( user ) {
+	const timeoutMs = 1500;
+	const loggedIn = user.get() !== false;
+	const ipc = require( 'electron' ).ipcRenderer;
+
+	const promiseTimeout = ( ms, promise ) => {
+		const timeout = new Promise( ( _, reject ) => {
+			const id = setTimeout( () => {
+				clearTimeout( id );
+				reject( `Request timed out in ${ ms } ms` );
+			}, ms );
+		} );
+
+		return Promise.race( [ promise, timeout ] );
+	};
+
+	const renderPromise = () => {
+		return new Promise( function ( resolve ) {
+			const sendUserAuth = () => {
+				debug( 'Sending user info to desktop...' );
+				ipc.send( 'user-auth', user, getToken() );
+			};
```

Comment:
You can call `ipc.send( 'user-auth' )` here directly instead of creating a function and call it later.

It's safe to add the `ipc.on( 'cookie-auth-complete' )` listener after the `ipc.send( 'user-auth' )` already finished. The only thing that needs to be guaranteed is that both calls are done synchronously in the same event loop tick.

`ipc.send( 'user-auth' )` doesn't send the IPC message immediately, before returning. It merely posts a task to the event loop. And that task is performed after the current task (execute a JS function) is complete.

Almost all async APIs in browser and Node.js work like this.